Omr installation docs (#39593)
openshift/openshift-docs: modules/mirror-registry-introduction.adoc

The mirror registry for Red Hat Openshift provides a pre-determined network configuration and reports deployed component credentials and access URLs upon success. A limited set of optional configuration inputs like fully qualified domain name (FQDN) services, superuser name and password, and custom TLS certificates are also provided. This provides users with a container registry so that they can easily create an offline mirror of all {product-title} release content when running {product-title} in restricted network environments.

The mirror registry for Red Hat Openshift is limited to hosting images that are required to install a disconnected {product-title} cluster, such as Release images or Red Hat Operator images. It uses local storage on your {op-system-base-full} machine, and storage supported by {op-system-base} is supported by the mirror registry for Red Hat Openshift. Content built by customers should not be hosted by the mirror registry for Red Hat Openshift.

> Unlike Red Hat Quay, the mirror registry for Red Hat Openshift is not a highly-available registry and only local filesystem storage is supported. Although it is technically possible, using the mirror registry for Red Hat Openshift with more than one cluster is discouraged, because multiple clusters can create a single point of failure when updating your cluster fleet. It is advised to leverage the mirror registry for Red Hat Openshift to install a cluster that can host a production-grade, highly-available registry such as Red Hat Quay, which can serve {product-title} content to other clusters.

Reviewer: ```suggestion
Unlike Red Hat Quay, the mirror registry for Red Hat Openshift is not a highly-available registry and only local file system storage is supported. Using the mirror registry for Red Hat Openshift with more than one cluster is discouraged, because multiple clusters can create a single point of failure when updating your cluster fleet. It is advised to leverage the mirror registry for Red Hat Openshift to install a cluster that can host a production-grade, highly-available registry such as Red Hat Quay, which can serve {product-title} content to other clusters.
```

> Use of the mirror registry for Red Hat Openshift is optional if another container registry is already available in the install environment.

Reviewer: If you do end up pulling content from the assembly into this module as suggested above, this line can probably go away

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-introduction_{context}"]
== Mirror registry for {product-title} introduction

> 

Reviewer 2: ```suggestion
For disconnected deployments of {product-title} a container registry is required to carry out the installation of the clusters. If users want to run a production-grade registry service on such a cluster a separate registry deployment is required to install the first cluster. The mirror registry addresses this need and is included in every OpenShift subscription. It is available for download in an online and fully independent offline version here: INSERT DOWNLOAD LINK
```

> Mirror registry allows users to install a small-scale version of Red Hat Quay and its required components using the `mirror-registry` command line interface (CLI) tool. Mirror registry comes equipped with pre-configured local storage and a local database. It also includes pre-configured user credentials and access permissions with a single set of inputs and no additional configuration choices to get started.

Reviewer 2: ```suggestion
The mirror registry allows users to install a small-scale version of Red Hat Quay and its required components using the `mirror-registry` command line interface (CLI) tool. The mirror registry will be deployed fully automatically with pre-configured local storage and a local database. It also includes auto-generated user credentials and access permissions with a single set of inputs and no additional configuration choices to get started.
```

Author: Can't use 'will' (future tense) in OCP docs. Changing to 'The mirror registry is deployed. . . "

> Mirror registry provides a pre-determined network configuration and reports deployed component credentials and URIs upon success. A limited set of optional configuration inputs like fully qualified domain name (FQDN) services, superuser name and password, and custom TLS certificates are also provided. These services allow users to easily create an offline mirror of all {product-title} release content when running {product-title} in restricted network environments.

Author: committed with slight alteration

> Mirror registry is limited to hosting images that are required to install a disconnected {product-title} cluster. It uses local storage on your {op-system-base-full} machine, and storage supported by {op-system-base} is supported by mirror registry. Mirror registry is not supported for production load, and is intended for use with {product-title} images only.

Reviewer 2: ```suggestion
The mirror registry is limited to hosting images that are required to install a disconnected {product-title} cluster. It uses local storage on your {op-system-base-full} machine, and storage supported by {op-system-base} is supported by mirror registry. The mirror registry is not supported for production load or custom application images (except for infrastructure-related monitoring and agent software), and is intended for use with {product-title} images only.
```

Author: Slightly altered to avoid parenthetical.

> 

Reviewer 2: ```suggestion
The mirror registry is based on a streamlined, minimal Red Hat Quay installation. Unlike Red Hat Quay, the mirror registry is not a highly available registry and only local filesystem storage is supported. Albeit technically possible, it is discouraged to use the the mirror registry with more than one cluster, to avoid create a single point of failure when updating your cluster fleet. It is advised to leverage the mirror registry to install a cluster which in turns hosts a production-grade, highly-available registry (such as Red Hat Quay) which serves {product-title} content to other clusters. Use of the mirror registry is optional if another container registry is already available in the install environment.
```

Author: Added, slightly altered.

// Module included in the following assemblies:
//
// * installing/installing-mirroring-installation-images.adoc

[id="mirror-registry-introduction_{context}"]
== Mirror registry for {product-title} introduction

> For disconnected deployments of {product-title}, a container registry is required to carry out the installation of the clusters. If users want to run a production-grade registry service on such a cluster, a separate registry deployment is required to install the first cluster. The mirror registry addresses this need and is included in every OpenShift subscription. It is available for download on the {product-title} link:https://console.redhat.com/openshift/downloads[*Downloads*] page.

Reviewer 3: We should talk with RCM, and possibly alter the 'link' here; as I think we can provide a customer portal link as well.

Unlike Red Hat Quay, the mirror registry is not a highly-available registry and only local filesystem storage is supported. Although it is technically possible, using the mirror registry with more than one cluster is discouraged, because multiple clusters can create a single point of failure when updating your cluster fleet. It is advised to leverage the mirror registry to install a cluster that can host a production-grade, highly-available registry such as Red Hat Quay, which can serve {product-title} content to other clusters.

Use of the mirror registry is optional if another container registry is already available in the install environment.

The mirror registry is comprised of the following images, which can be found in the link:https://catalog.redhat.com/software/containers/explore[Red Hat Ecosystem Catalog]:

* Quay: registry.redhat.io/quay/quay-rhel8:v3.6.1
* Redis: registry.redhat.io/rhel8/redis-6:1-25
* Postgres: registry.redhat.io/rhel8/postgresql-10:1-161

> * Mirror registry: quay.io/quay/mirror-registry-ee:latest

Reviewer 2: Wondering whether we need to link these images. They are all part of the payload. Do we need this? We might confuse users into having to download these.
